ge/org.osate.ge/src/org/osate/ge/internal/ui/tools/ToolUtil.java

```diff
@@ -80,4 +107,181 @@ public static Context findContextExcludeOwner(final BusinessObjectContext boc,
 		final BusinessObjectContext contextBoc = findContextAncestorBoc(boc);
 		return contextBoc == null || contextBoc == ownerBoc ? null : (Context) contextBoc.getBusinessObject();
 	}
+
+	/**
+	 * Get the errors and warnings for referenced elements' root
+	 */
+	public static List<Diagnostic> getAllReferencedPackageDiagnostics(final BusinessObjectContext boc) {
+		// Find root to get all referenced package diagnostics
+		final BusinessObjectContext root = findRoot(boc)
+				.orElseThrow(() -> new RuntimeException("Cannot find root"));
+
+		// Get get root elements
+		final Set<NamedElement> rootElements = root.getAllDescendants().map(queryable -> {
+			final Object bo = queryable.getBusinessObject();
+			return Element.class.isInstance(bo) ? bo : null;
+		}).filter(Predicates.notNull()).map(bo -> ((Element) bo).getElementRoot()).collect(Collectors.toSet());
```

Comment:
Filter out root elements which are null. Could filter out all those that are not packages.